vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc

```diff
@@ -238,6 +255,22 @@ void Promotion::OnGetAllPromotions(
       callback);
 }
 
+void Promotion::OnGetAllPromotionsFromDatabase(
+    ledger::PromotionMap promotions,
+    ledger::FetchPromotionCallback callback) {
+  HandleExpiredPromotions(ledger_, &promotions);
+
+  ledger::PromotionList promotions_ui;
```

Comment:
maybe we can extract this into anon namespace above so both paths have the same function for returning ui promotions, this way if we for example change what to return (in that if, new status for example) we only change it on one place and is less error prone

Reply:
Makes sense, done.